ColorPatterns/Scenes/Game/GameSounds.swift

```diff
@@ -8,20 +8,52 @@
 
 import AVFoundation
 
-struct GameSounds {
-    
-    private var audioPlayer: AVAudioPlayer?
-
-    private let soundsArray = [K.Sounds.note1, K.Sounds.note2, K.Sounds.note3, K.Sounds.note4, K.Sounds.note5]
-    
-    mutating func play(soundFile: Int) {
-        guard let soundURL = Bundle.main.url(forResource: soundsArray[soundFile],
-                                             withExtension: K.Sounds.wavFormat) else { return }
+final class GameSounds {
+
+    private var soundFile: Int = 0
+    private var url: URL? {
+        let soundURL = Bundle.main.url(forResource: soundsArray[soundFile],
+                                       withExtension: Constants.Sounds.wavFormat)
+        return soundURL
+    }
+    //FIXME: That doesn't work properly.
+    // Only some sounds are played because only one player is created and it has to play sound to the end
+    private lazy var audioPlayer: AVAudioPlayer = {
+        [unowned self] in
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/vandermesis/ColorPatterns/blob/c279b1ccceb81114848aa1e590426203038db92d/ColorPatterns/Scenes/Game/GameSounds.swift#L22"></span>It’s safer to use weak instead of unowned</td>
    </tr>
  </tbody>
</table>